framework/DataObjects/DataSet.py

```diff
@@ -197,16 +197,15 @@ def addRealization(self, rlz):
     #
     #  Yours truly, talbpw, May 2019
     #########
-    # protect against back-changing realization
-    rlz = copy.deepcopy(rlz)
     # if index map was included, remove that now before checking variables
-    indexMap = rlz.pop('_indexMap', None)
+    indexMap = rlzIn.get('_indexMap', None)
```

Comment:
I'm shocked this has no downstream ramifications. I guess the index map doesn't get considered as a variable because we do the selective reconstruction below? Given the tests pass, it must not be a big deal.

Reply:
Yes, you are right, we do the reconstruction right after the check of the indexMap, and it will not affect other operations on the realizations.